packages/aws-cdk-lib/core/lib/stack.ts

```diff
@@ -1644,9 +1649,14 @@ export function rootPathTo(construct: IConstruct, ancestor?: IConstruct): IConst
  * has only one component. Otherwise we fall back to the regular "makeUniqueId"
  * behavior.
  */
-function makeStackName(components: string[]) {
-  if (components.length === 1) { return components[0]; }
-  return makeUniqueResourceName(components, { maxLength: 128 });
+function makeStackName(components: string[], prefix: string='') {
+  if (components.length === 1) {
+    const stack_name = prefix + components[0];
```

Comment:
why not just let `makeUniqueResourceName` do this?

Reply:
This is done also to make the stack names to be as close as possible to the original behavior. If we go through the makeUniqueResourceName, we would lose all non allowed special character (dashes for example) from the stack name if the component has any in its name. So we don't want single component stacks to go through the makeUniqueResourceName in order to preserve its original name.